From 3a4087e836cb111ff9ce6e6764ec31691c342a09 Mon Sep 17 00:00:00 2001
From: Victor Gu <xigu@marvell.com>
Date: Tue, 9 May 2017 14:48:57 +0800
Subject: [PATCH 0997/1345] vfix: usb: gadget: u3d: remove USB3 reset during
 disconnection

commit  6011f33204550a1e88ac80f679ad0832134e831c from
https://github.com/MarvellEmbeddedProcessors/linux-marvell.git

The routine mvc2_hw_reset() is used to do whole USB3 device controller
reset, which should be used after VBUS is on. When USB3 VBUS is on,
it means USB3 device is plugged into host, driver should prepare to
communicate with host, and the reset is mandatory.
If VBUS is off, it means device is disconnected from host and the
USB3 device functionality stops, thus it is not necessary to do reset
during VBUS disconnection.
This patch removes the reset during USB3 device VBUS disconnection.
The USB3 device suspend and resume long run instability issue is fixed
after removing this reset during disconnection.
JIRA A3700-959.

Change-Id: Idb506c9392e35f3adf81d375c31a67c49358b6b9
Signed-off-by: Victor Gu <xigu@marvell.com>
Reviewed-on: http://vgitil04.il.marvell.com:8080/39251
Tested-by: iSoC Platform CI <ykjenk@marvell.com>
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 drivers/usb/gadget/udc/mvebu_u3d.c |    1 -
 1 file changed, 1 deletion(-)

diff --git a/drivers/usb/gadget/udc/mvebu_u3d.c b/drivers/usb/gadget/udc/mvebu_u3d.c
index 3abc330..6cae655 100644
--- a/drivers/usb/gadget/udc/mvebu_u3d.c
+++ b/drivers/usb/gadget/udc/mvebu_u3d.c
@@ -2293,7 +2293,6 @@ void mvc2_connect(struct mvc2 *cp, int is_on)
 		val = MV_CP_READ(cp->reg->global_control);
 		val &= ~MVCP_GLOBAL_CONTROL_SOFT_CONNECT;
 		MV_CP_WRITE(val, cp->reg->global_control);
-		mvc2_hw_reset(cp);
 	}
 }
 
-- 
1.7.9.5

